packages/astro/src/types/public/internal.ts

```diff
@@ -66,7 +66,7 @@ export interface RouteData {
 	 * console.log(generate({ id: 'presentation' })) // will log `/blog/presentation`
 	 * ```
 	 */
-	generate: (data?: any) => string;
+	// generate: (data?: any) => string;
```

Comment:
Is this meant to be removed?

Reply:
Yes. We remove it from `RouteData` altogether.